src/libmongoc/tests/test-mongoc-retryable-writes.c

```diff
@@ -568,6 +568,236 @@ test_unsupported_storage_engine_error (void)
    mongoc_client_destroy (client);
    mock_rs_destroy (rs);
 }
+
+/* The following 3 tests check that the original reply and error is returned
+ * after encountering an error with a RetryableWriteError and a
+ * NoWritesPerformed label. The tests use the same callback function.
+ *
+ * Each test checks a different code path for retryable writes:
+ * mongoc_collection_insert_one, mongoc_client_command_simple, and
+ * mongoc_collection_find_and_modify_with_opts
+ *
+ * These tests require a >=6.0 replica set.
+ */
+typedef struct {
+   mongoc_client_t *client;
+   bool configure_second_fail;
+   char *failCommand;
+} prose_test_3_apm_ctx_t;
+
+static void
+prose_test_3_on_command_success (const mongoc_apm_command_succeeded_t *event)
+{
+   bson_iter_t iter;
+   bson_error_t error;
+   const bson_t *reply = mongoc_apm_command_succeeded_get_reply (event);
+   prose_test_3_apm_ctx_t *ctx =
+      mongoc_apm_command_succeeded_get_context (event);
+
+   // wait for a writeConcernError and then set a second failpoint
+   if (bson_iter_init_find (&iter, reply, "writeConcernError") &&
+       ctx->configure_second_fail) {
+      ctx->configure_second_fail = false;
+      ASSERT_OR_PRINT (
+         mongoc_client_command_simple (
+            ctx->client,
+            "admin",
+            tmp_bson (
+               "{'configureFailPoint': 'failCommand', 'mode': {'times': 1},"
+               " 'data': { 'failCommands': ['%s'], 'errorCode': "
+               "10107, "
+               "'errorLabels': ['RetryableWriteError', 'NoWritesPerformed']}}",
+               ctx->failCommand),
+            NULL,
+            NULL,
+            &error),
+         error);
+   }
+}
+
+uint32_t
```

Comment:
```suggestion
static uint32_t
```

Prefer marking helper functions `static`, since they are not needed by other files.